tests/study_tests/test_study.py

```diff
@@ -990,3 +990,43 @@ def test_tell_pruned_values() -> None:
 
     study.tell(trial, state=TrialState.PRUNED)
     assert study.trials[-1].value is None
+
+
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_trial_duration_calculation(storage_mode: str) -> None:
+
+    with StorageSupplier(storage_mode) as storage:
+        study = create_study(storage=storage)
+
+        def objective(trial: Trial) -> float:
+            time.sleep(1)
+            x = trial.suggest_int("x", -10, 10)
+            return x
+
+        study.enqueue_trial(params={"x": 1})
```

Comment:
How about adding an assertion that `datetime_start` is `None` after this line?

Reply:
Added.